cmd/generator/main.go

```diff
@@ -177,45 +177,31 @@ func getReleasePlugins(ctx context.Context, client *github.Client, repositoryNam
 	}
 
 	var plugins []*model.Plugin
-	// Keep track of the latest plugin compatible with the given server version
-	minServerVersionsSeen := map[string]*model.Plugin{}
+	var latestPlugin *model.Plugin
 	for _, release := range releases {
-		releasePlugin, err := getReleasePlugin(release, repository, existingPlugins)
+		plugin, err := getReleasePlugin(release, repository, existingPlugins)
 		if err != nil {
 			return nil, errors.Wrapf(err, "failed to get release plugin for %s", release.GetName())
 		}
 
-		if releasePlugin == nil {
+		if plugin == nil {
 			logger.Warnf("no plugin found for release %s", release.GetName())
 			continue
 		}
 
-		if minServerVersionsSeen[releasePlugin.Manifest.MinServerVersion] != nil {
-			if releasePlugin.Manifest.Version == "" {
-				return nil, errors.Errorf("version is empty for manifest.Id %s", releasePlugin.Manifest.Id)
-			}
-
-			lastSeenPlugin := minServerVersionsSeen[releasePlugin.Manifest.MinServerVersion]
-			lastSeenPluginVersion, err := semver.Parse(lastSeenPlugin.Manifest.Version)
-			if err != nil {
-				return nil, errors.Wrapf(err, "failed to parse version %s", lastSeenPlugin.Manifest.Version)
-			}
-
-			releasePluginVersion, err := semver.Parse(releasePlugin.Manifest.Version)
-			if err != nil {
-				return nil, errors.Wrapf(err, "failed to parse release plugin version %s", releasePlugin.Manifest.Version)
-			}
+		if plugin.Manifest.Version == "" {
+			return nil, errors.Errorf("version is empty for manifest.Id %s", plugin.Manifest.Id)
+		}
 
-			// Ignore if we have the latest plugin version for this server version
-			if lastSeenPluginVersion.GTE(releasePluginVersion) {
-				continue
-			}
+		pluginVersion, err := semver.Parse(plugin.Manifest.Version)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to parse release plugin version %s", plugin.Manifest.Version)
 		}
 
-		minServerVersionsSeen[releasePlugin.Manifest.MinServerVersion] = releasePlugin
-	}
+		if latestPlugin == nil || pluginVersion.GT(semver.MustParse(latestPlugin.Manifest.Version)) {
+			latestPlugin = plugin
```

Comment:
What is `latestPlugin` used for?

Reply:
Good catch: this was from a version of my changes where I tracked the latest version of a plugin for handling changes of plugin ids from the same repository, but this become obsolete once we eliminated pre-release plugins, so I abandoned the extra complexity... but forgot to remove it here :)

Fixed!